src/main/java/org/craftercms/engine/service/context/SiteContextManager.java

```diff
@@ -187,6 +204,16 @@ public void destroyContext(String siteName) {
     public SiteContext getContext(String siteName, boolean fallback) {
         SiteContext siteContext = contextRegistry.get(siteName);
         if (siteContext == null) {
+            if(!fallbackSite.equals(siteName)) {
+                try {
+                    entitlementValidator.validateEntitlement(Module.ENGINE, EntitlementType.SITE,
+                        (int) contextRegistry.entrySet().stream()
+                                                        .filter(entry -> !fallbackSite.equals(entry.getKey()))
+                                                        .count(), 1);
+                } catch (EntitlementException e) {
+                    return null;
```

Comment:
the `EntitlementValidator` takes care of all logging so it's unified and modules just need to handle exceptions in their own way (for example studio shows a popup), let me know if engine should do something different than just saying it can't load the site

Reply:
No, it makes sense now. Thanks